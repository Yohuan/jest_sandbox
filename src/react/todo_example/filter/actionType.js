const SET_FILTER = 'FILTER/SET';

export {
  SET_FILTER,
};
