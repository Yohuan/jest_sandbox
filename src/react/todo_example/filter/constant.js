const filterType = {
  ALL: 'all',
  COMPLETED: 'completed',
  UNCOMPLETED: 'uncompleted',
};

export {
  filterType,
};
