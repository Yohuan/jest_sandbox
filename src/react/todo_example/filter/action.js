import { SET_FILTER } from './actionType';

const setFilter = filterType => ({
  type: SET_FILTER,
  filterType,
});

export {
  setFilter,
};
