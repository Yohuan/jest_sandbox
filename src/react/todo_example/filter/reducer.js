import { SET_FILTER } from './actionType';
import { filterType } from './constant';

export default (state = filterType.ALL, action) => {
  switch (action.type) {
    case SET_FILTER: {
      return action.filterType;
    }
    default: {
      return state;
    }
  }
};
