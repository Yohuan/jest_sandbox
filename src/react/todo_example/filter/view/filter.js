import React from 'react';

import { Link as FilterLink } from './link';
import { filterType } from '../constant';

const Filter = () => (
  <p className="filter">
    <FilterLink filter={filterType.ALL}>{filterType.ALL}</FilterLink>
    <FilterLink filter={filterType.COMPLETED}>{filterType.COMPLETED}</FilterLink>
    <FilterLink filter={filterType.UNCOMPLETED}>{filterType.UNCOMPLETED}</FilterLink>
  </p>
);

export {
  Filter,
};
