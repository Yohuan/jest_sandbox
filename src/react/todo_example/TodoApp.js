import React from 'react';

import { view as filterView } from './filter';
import { view as todosView } from './todos';

const TodoApp = () => (
  <div className="todo-app">
    <todosView.Todos />
    <filterView.Filter />
  </div>
);

export {
  TodoApp,
};
