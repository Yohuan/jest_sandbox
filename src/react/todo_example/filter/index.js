import * as constant from './constant';
import reducer from './reducer';
import { Filter } from './view/filter';

const view = {
  Filter,
};

export {
  constant,
  reducer,
  view,
};
