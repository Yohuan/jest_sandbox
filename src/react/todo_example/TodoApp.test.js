import React from 'react';
import Immutable from 'immutable';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { mount } from 'enzyme';

import { TodoApp } from './TodoApp';
import { constant as filterConstant } from './filter';

describe('snapshot test', () => {
  const initialStoreState = Immutable.Map({
    todos: [
      {
        id: 0,
        text: 'write tests',
        completed: true,
      },
      {
        id: 1,
        text: 'write more tests',
        completed: false,
      },
    ],
    filter: filterConstant.filterType.ALL,
  });
  const fakeRuducer = state => (state);
  const store = createStore(fakeRuducer, initialStoreState);
  const subject = (
    <Provider store={store}>
      <TodoApp />
    </Provider>
  );

  test('<TodoApp /> should render some todos', () => {
    const wrapper = mount(subject);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
