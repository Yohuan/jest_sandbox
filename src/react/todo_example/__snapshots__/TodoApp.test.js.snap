// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test <TodoApp /> should render some todos 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <TodoApp>
    <div
      className="todo-app"
    >
      <Connect(_Todos)>
        <_Todos
          addTodo={[Function]}
          removeTodo={[Function]}
          todos={
            Array [
              Object {
                "completed": true,
                "id": 0,
                "text": "write tests",
              },
              Object {
                "completed": false,
                "id": 1,
                "text": "write more tests",
              },
            ]
          }
          toggleTodo={[Function]}
        >
          <div
            className="todos"
          >
            <AddTodo
              onAdd={[Function]}
            >
              <div
                className="add-todo"
              >
                <form
                  onSubmit={[Function]}
                >
                  <input
                    className="new-todo"
                  />
                  <button
                    className="add-btn"
                    type="submit"
                  >
                    Add
                  </button>
                </form>
              </div>
            </AddTodo>
            <TodoList
              onRemoveTodo={[Function]}
              onToggleTodo={[Function]}
              todos={
                Array [
                  Object {
                    "completed": true,
                    "id": 0,
                    "text": "write tests",
                  },
                  Object {
                    "completed": false,
                    "id": 1,
                    "text": "write more tests",
                  },
                ]
              }
            >
              <ul
                className="todo-list"
              >
                <TodoItem
                  completed={true}
                  key="0"
                  onRemove={[Function]}
                  onToggle={[Function]}
                  text="write tests"
                >
                  <li
                    className="todo-item"
                    style={
                      Object {
                        "textDecoration": "line-through",
                      }
                    }
                  >
                    <input
                      checked="checked"
                      className="toggle"
                      onClick={[Function]}
                      readOnly={true}
                      type="checkbox"
                    />
                    write tests
                    <button
                      className="remove-btn"
                      onClick={[Function]}
                      type="button"
                    >
                      X
                    </button>
                  </li>
                </TodoItem>
                <TodoItem
                  completed={false}
                  key="1"
                  onRemove={[Function]}
                  onToggle={[Function]}
                  text="write more tests"
                >
                  <li
                    className="todo-item"
                    style={
                      Object {
                        "textDecoration": "none",
                      }
                    }
                  >
                    <input
                      checked=""
                      className="toggle"
                      onClick={[Function]}
                      readOnly={true}
                      type="checkbox"
                    />
                    write more tests
                    <button
                      className="remove-btn"
                      onClick={[Function]}
                      type="button"
                    >
                      X
                    </button>
                  </li>
                </TodoItem>
              </ul>
            </TodoList>
          </div>
        </_Todos>
      </Connect(_Todos)>
      <Filter>
        <p
          className="filter"
        >
          <Connect(_Link)
            filter="all"
          >
            <_Link
              active={true}
              filter="all"
              onClick={[Function]}
            >
              <button
                className="filter-btn"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "#008CBA",
                    "color": "white",
                  }
                }
                type="button"
              >
                all
              </button>
            </_Link>
          </Connect(_Link)>
          <Connect(_Link)
            filter="completed"
          >
            <_Link
              active={false}
              filter="completed"
              onClick={[Function]}
            >
              <button
                className="filter-btn"
                onClick={[Function]}
                style={null}
                type="button"
              >
                completed
              </button>
            </_Link>
          </Connect(_Link)>
          <Connect(_Link)
            filter="uncompleted"
          >
            <_Link
              active={false}
              filter="uncompleted"
              onClick={[Function]}
            >
              <button
                className="filter-btn"
                onClick={[Function]}
                style={null}
                type="button"
              >
                uncompleted
              </button>
            </_Link>
          </Connect(_Link)>
        </p>
      </Filter>
    </div>
  </TodoApp>
</Provider>
`;
